node/db/src/rolling/gc.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//!
+//! The current implementation of the garbage collector is a concurrent,
+//! semi-space one.
+//!
+//! ## GC workflow
+//! 1. Walk back from the current heaviest tipset to the genesis block, collect
+//! all the blocks that are reachable from the snapshot
+//! 2. writes blocks that are absent from the `current` database to it
+//! 3. delete `old` database(s)
+//! 4. sets `current` database to a newly created one
+//!
+//! ## Correctness
+//! This alroghrim considers all blocks that are visited during the snapshot
```

Comment:
algorithm

Reply:
Good catch! But why is it not caught by the spellcheck CI task

Reply:
Hmm, something is wrong with the spellchecker...